src/OpenTelemetry.Instrumentation.AspNetCore/MeterProviderBuilderExtensions.cs

```diff
@@ -28,23 +29,47 @@ public static class MeterProviderBuilderExtensions
         /// Enables the incoming requests automatic data collection for ASP.NET Core.
         /// </summary>
         /// <param name="builder"><see cref="MeterProviderBuilder"/> being configured.</param>
+        /// <param name="configureAspNetCoreInstrumentationOptions">ASP.NET Core Request configuration options.</param>
         /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
         public static MeterProviderBuilder AddAspNetCoreInstrumentation(
-            this MeterProviderBuilder builder)
+            this MeterProviderBuilder builder,
+            Action<AspNetCoreInstrumentationOptions> configureAspNetCoreInstrumentationOptions = null)
         {
             Guard.ThrowIfNull(builder);
 
-            // TODO: Implement an IDeferredMeterProviderBuilder
-
             // TODO: Handle AspNetCoreInstrumentationOptions
-            //   Filter - makes sense for metric instrumentation
             //   Enrich - do we want a similar kind of functionality for metrics?
             //   RecordException - probably doesn't make sense for metric instrumentation
-            //   EnableGrpcAspNetCoreSupport - this instrumentation will also need to also handle gRPC requests
+            //   EnableGrpcAspNetCoreSupport - this instrumentation will also need to handle gRPC requests
+
+            if (builder is IDeferredMeterProviderBuilder deferredMeterProviderBuilder)
+            {
+                return deferredMeterProviderBuilder.Configure((sp, builder) =>
+                {
+                    AddAspNetCoreInstrumentation(builder, sp.GetOptions<AspNetCoreInstrumentationOptions>(), configureAspNetCoreInstrumentationOptions);
+                });
+            }
+
+            return AddAspNetCoreInstrumentation(builder, new AspNetCoreInstrumentationOptions(), configureAspNetCoreInstrumentationOptions);
+        }
 
-            var instrumentation = new AspNetCoreMetrics();
+        internal static MeterProviderBuilder AddAspNetCoreInstrumentation(
+            this MeterProviderBuilder builder,
+            AspNetCoreMetrics instrumentation)
+        {
             builder.AddMeter(AspNetCoreMetrics.InstrumentationName);
             return builder.AddInstrumentation(() => instrumentation);
         }
+
+        private static MeterProviderBuilder AddAspNetCoreInstrumentation(
+            MeterProviderBuilder builder,
+            AspNetCoreInstrumentationOptions options,
+            Action<AspNetCoreInstrumentationOptions> configure = null)
+        {
+            configure?.Invoke(options);
+            return AddAspNetCoreInstrumentation(
```

Comment:
This is where the discrepancy is between traces and metrics - in traces we will new-up `HttpInLIstener` here and pass it to the instrumentation (`AspNetCoreMetrics` in this case). Here it is a little bit more complicated, because metrics listener also expects the `Meter` instance, which is disposable. Currently `AspNetCoreMetrics` will maintain meter's lifecycle, and I didn't want to push it out to here, which will be hard to do, given that this is a static POP by itself. So I just pass in options to the instrumentation class, and it passes them to the listener. Should not be a big deal, just one thing to note.